packages/playwright-core/src/server/trace/recorder/tracing.ts

```diff
@@ -496,8 +497,10 @@ export class Tracing extends SdkObject implements InstrumentationListener, Snaps
 function visitTraceEvent(object: any, sha1s: Set<string>): any {
   if (Array.isArray(object))
     return object.map(o => visitTraceEvent(o, sha1s));
+  if (object instanceof Dispatcher)
```

Comment:
when does this happen?

Reply:
When we call `_appendTraceEvent` (after) for `newPage`. This returns `{ result: PageDispatcher }`.